room.cpp

```diff
@@ -169,11 +169,32 @@ void Room::setLastReadEvent(User* user, QString eventId)
     emit lastReadEventChanged(user);
 }
 
-void Room::markMessageAsRead(Event* event)
+void Room::markMessagesAsRead(Timeline::const_iterator end)
 {
-    setLastReadEvent(connection()->user(), event->id());
-    if (event->senderId() != connection()->userId())
-        d->connection->postReceipt(this, event);
+    if (messageEvents().empty())
+        return;
+
+    QString prevLastReadId = lastReadEvent(connection()->user());
+    setLastReadEvent(connection()->user(), (*--end)->id());
```

Comment:
Aw, I'm terribly sorry. That was documentation written before implementation :( Basically, the way the iterator is passed now allows me to send an iterator even into an empty container without double-checking for emptiness. But I agree that it's a fairly thin benefit in this particular case, and using the STL-style `end` iterator looks kind of clumsy.